src/vos/vos_tree.c

```diff
@@ -854,7 +854,7 @@ tree_open_create(struct vos_object *obj, enum vos_tree_class tclass, int flags,
 		goto out;
 	}
 
-	if (flags & SUBTR_EVT) {
+	if (flags & SUBTR_EVT && expected_flag == KREC_BF_EVT) {
```

Comment:
I supose this code is only for only, see following "goto out" codes, L875 won't be called to use flags though, i could remove SUBTR_EVT from flags if you think it more clean.

Reply:
To be clear, the create case, should use SUBTR_EVT because it's controlled by the flag that we can elide the akey (or is under the akey).   For the open case, we need to check what the current df flag is and the code is incorrectly ignoring what we determined it should be.   So just the 2nd check should suffice.  Good catch.